src/awscdk/github.ts

```diff
@@ -311,40 +305,72 @@ export class GithubCDKPipeline extends CDKPipeline {
             packages: JobPermission.READ,
           },
         },
-        steps: [{
-          name: 'Checkout',
-          uses: 'actions/checkout@v4',
-        }, {
-          name: 'AWS Credentials',
-          uses: 'aws-actions/configure-aws-credentials@master',
-          with: {
-            'role-to-assume': this.options.iamRoleArns?.deployment?.[stage.name] ?? this.options.iamRoleArns?.default,
-            'role-session-name': 'GitHubAction',
-            'aws-region': stage.env.region,
+        steps: [
+          {
+            name: 'Checkout',
+            uses: 'actions/checkout@v4',
           },
-        }, {
-          uses: 'actions/download-artifact@v4',
-          with: {
-            name: 'cloud-assembly',
-            path: `${this.app.cdkConfig.cdkout}/`,
-          },
-        },
-        ...preInstallSteps.flatMap(s => s.steps),
-        ...this.renderInstallCommands().map(cmd => ({
-          run: cmd,
-        })),
-        ...this.renderDeployCommands(stage.name).map(cmd => ({
-          run: cmd,
-        })),
-        {
-          uses: 'actions/upload-artifact@v3',
-          with: {
-            name: `cdk-outputs-${stage.name}`,
-            path: `cdk-outputs-${stage.name}.json`,
-          },
-        }],
+          ...steps.flatMap(s => s.steps),
+        ],
       });
       this.deploymentStages.push(stage.name);
     }
   }
+
+  /**
+   * Creates a job to deploy the CDK application to AWS.
+   * @param stage - The independent stage to create.
+   */
+  public createIndependentDeployment(stage: IndependentStage): void {
+    const steps = [
+      new AwsAssumeRoleStep(this.project, {
+        roleArn: this.options.iamRoleArns?.deployment?.[stage.name] ?? this.options.iamRoleArns?.default!,
+        region: stage.env.region,
+      }),
+      ...this.options.preInstallSteps ?? [],
+      new SimpleCommandStep(this.project, this.renderInstallCommands()),
+
+      ...this.options.preSynthSteps ?? [],
+      new SimpleCommandStep(this.project, this.renderSynthCommands()),
+      ...this.options.postSynthSteps ?? [],
+
+      new SimpleCommandStep(this.project, this.renderDiffCommands(stage.name)),
+      ...stage.postDiffSteps ?? [],
+
+      new SimpleCommandStep(this.project, this.renderDeployCommands(stage.name)),
+      ...stage.postDeploySteps ?? [],
+
+      new UploadArtifactStep(this.project, {
+        name: `cdk-outputs-${stage.name}`,
+        path: `cdk-outputs-${stage.name}.json`,
+      }),
+    ].map(s => s.toGithub());
+
+    // Create new workflow for deployment
+    const stageWorkflow = this.app.github!.addWorkflow(`deploy-${stage.name}`);
```

Comment:
this workflow got the "auto-run on push" according to the snapshot that is checked in - do we really want that? 

or should it be a workflow that we "manually" trigger only?

Reply:
see below